cli_tools/gce_vm_image_import/importer/request.go

```diff
@@ -0,0 +1,156 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package importer
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+	"time"
+
+	daisyutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/daisycommon"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+// Flags that are validated.
+const (
+	ImageFlag          = "image_name"
+	ClientFlag         = "client_id"
+	BYOLFlag           = "byol"
+	DataDiskFlag       = "data_disk"
+	OSFlag             = "os"
+	CustomWorkflowFlag = "custom_translate_workflow"
+)
+
+func (args *ImageImportRequest) validate() error {
+
+	if err := args.checkRequiredArguments(); err != nil {
+		return err
+	}
+
+	if args.BYOL && (args.DataDisk || args.OS != "" || args.CustomWorkflow != "") {
+		return fmt.Errorf("when -%s is specified, -%s, -%s, and -%s have to be empty",
+			BYOLFlag, DataDiskFlag, OSFlag, CustomWorkflowFlag)
+	}
```

Comment:
This might not be an issue, depending on how it's integrated with OVF import, but some of these flag names don't exist in OVF import and they use a different notation ("-" instead of "_" to separate words in flag names.

I assume these would be panics in OVF Import, as it means OVF importer didn't set the request properly?